### PR TITLE
Add Krystal Wallet adapter

### PR DESCRIPTION
**Pull request by tungnguyen20**
Add adapter for our defi wallet app https://krystal.app 
Currently, we only support in-app browser for android and iOS. Chrome will be soon.

### CODE REVIEW

File: packages/wallets/krystal/src/adapter.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    WalletName,
+    WalletNotConnectedError,
+    WalletNotReadyError,
+    WalletPublicKeyError,
+    WalletReadyState,
+    WalletSignMessageError,
+    WalletSignTransactionError,
+} from '@solana/wallet-adapter-base';
+import { PublicKey, Transaction } from '@solana/web3.js';
+import bs58 from 'bs58';
```
Comment (jordansexton) [inline packages/wallets/krystal/src/adapter.ts:14]: This import isn't used, so you won't need the dependency either :)

File: packages/wallets/krystal/package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    },
+    "dependencies": {
+        "@solana/wallet-adapter-base": "^0.9.9",
+        "bs58": "^4.0.1"
+    },
+    "devDependencies": {
+        "@solana/web3.js": "^1.44.3",
+        "@types/bs58": "^4.0.1"
+    },
+    "gitHead": "389767182619fc76ab939920f75cc7194ff77340"
```
Comment (jordansexton) [inline packages/wallets/krystal/package.json:40]:
```suggestion
    }
```

This probably shouldn't be copied.

**Review by jordansexton — COMMENTED** [review]
Great PR, thanks! Only two minor fixes requested.

**Comment by tungnguyen20** [comment]
> Great PR, thanks! Only two minor fixes requested.

Just updated, please help me continue. Thank you :D

**Comment by jordansexton** [comment]
Published `@solana/wallet-adapter-krystal@0.1.0`, please test it out!

**Comment by tungnguyen20** [comment]
@jordansexton Sure. My team will continue our process. Thank you very much for your support.

**Pull request merged**